src/Deserializers/LegacyPropertyDeserializer.php

```diff
@@ -66,32 +72,26 @@ private function getDataTypeId() {
 		return $this->serialization['datatype'];
 	}
 
-	private function setPropertyId() {
-		if ( array_key_exists( 'entity', $this->serialization ) ) {
-			$this->property->setId( $this->getPropertyId() );
-		}
-	}
-
 	/**
-	 * @return PropertyId
 	 * @throws InvalidAttributeException
+	 * @return PropertyId|null
 	 */
 	private function getPropertyId() {
-		$id = $this->idDeserializer->deserialize( $this->serialization['entity'] );
+		if ( array_key_exists( 'entity', $this->serialization ) ) {
```

Comment:
This does not change semantics as far as I can tell. Look how the old code was arranged:
* `deserialize` called `setPropertyId` which called `getPropertyId`, but only if the array key is set.
* Now `deserialize` calls `getPropertyId`. The array key check still happens. The only difference is that it returns null instead of 'setId' not being called at all.